src/base/bittorrent/tracker.cpp

```diff
@@ -392,26 +392,27 @@ void Tracker::prepareAnnounceResponse(const TrackerAnnounceRequest &announceReq)
     };
 
     // peer list
-    // [BEP-7] IPv6 Tracker Extension (partial support)
+    // [BEP-7] IPv6 Tracker Extension (partial support - only the part that concerns BEP-23)
     // [BEP-23] Tracker Returns Compact Peer Lists
     if (announceReq.compact) {
-        lt::entry::list_type peerList;
-        lt::entry::list_type peer6List;
+        lt::entry::string_type peers;
+        lt::entry::string_type peers6;
```

Comment:
Could you explain what is the reason behind this change?

Reply:
See http://www.bittorrent.org/beps/bep_0023.html

> trackers may return peers as a packed string rather than as a bencoded list.

> The compact format uses the same peers key in the bencoded tracker response, but the value is a bencoded string rather than a bencoded list.

and the example response at http://www.bittorrent.org/beps/bep_0007.html (emphasis mine):

> d8:intervali1800e**5:peers6:iiiipp6:peers618:iiiiiiiiiiiiiiiipp**e

Notice how after `5:peers` there is a packed string of IPv4 endpoints, in this case with only 1 endpoint: `6:iiiipp`. Similarly, after `6:peers6` there is a packed string of IPv6 endpoints, in this case with only 1 endpoint as well: `18:iiiiiiiiiiiiiiiipp`. In both cases, `i` represents an address byte and `p` a port byte, in network byte order (i.e. big endian, "most significant byte first").